fmriprep/workflows/bold/registration.py

```diff
@@ -297,31 +299,47 @@ def init_bold_t1_trans_wf(freesurfer, mem_gb, omp_nthreads, use_fieldwarp=False,
             (aparc_t1w_tfm, outputnode, [('output_image', 'bold_aparc_t1')]),
         ])
 
-    # Merge transforms placing the head motion correction last
-    nforms = 2 + int(use_fieldwarp)
-    merge_xforms = pe.Node(niu.Merge(nforms), name='merge_xforms',
-                           run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
-    workflow.connect([
-        (inputnode, merge_xforms, [('hmc_xforms', 'in%d' % nforms)])
-    ])
-
-    if use_fieldwarp:
-        workflow.connect([
-            (inputnode, merge_xforms, [('fieldwarp', 'in2')])
-        ])
+    # merge 3D volumes into 4D timeseries
+    merge = pe.Node(Merge(compress=use_compression), name='merge', mem_gb=mem_gb)
 
     bold_to_t1w_transform = pe.Node(
         MultiApplyTransforms(interpolation="LanczosWindowedSinc", float=True, copy_dtype=True),
         name='bold_to_t1w_transform', mem_gb=mem_gb * 3 * omp_nthreads, n_procs=omp_nthreads)
 
-    merge = pe.Node(Merge(compress=use_compression), name='merge', mem_gb=mem_gb)
+    if not multiecho:
+        # Merge transforms placing the head motion correction last
+        nforms = 2 + int(use_fieldwarp)
+        merge_xforms = pe.Node(niu.Merge(nforms), name='merge_xforms',
+                               run_without_submitting=True, mem_gb=DEFAULT_MEMORY_MIN_GB)
+        if use_fieldwarp:
+            workflow.connect([
+                (inputnode, merge_xforms, [('fieldwarp', 'in2')])
+            ])
+
+        workflow.connect([
+            # merge transforms
+            (inputnode, merge_xforms, [
+                ('hmc_xforms', 'in%d' % nforms),
+                ('itk_bold_to_t1', 'in1')]),
+            (merge_xforms, bold_to_t1w_transform, [('out', 'transforms')]),
+            (inputnode, bold_to_t1w_transform, [('bold_split', 'input_image')]),
+        ])
+
+    else:
+        from nipype.interfaces.fsl import Split as FSLSplit
+        bold_split = pe.Node(FSLSplit(dimension='t'), name='bold_split',
+                             mem_gb=DEFAULT_MEMORY_MIN_GB)
+
+        workflow.connect([
+            (inputnode, bold_split, [('bold_split', 'in_file')]),
+            (bold_split, bold_to_t1w_transform, [('out_files', 'input_image')]),
+            (inputnode, bold_to_t1w_transform, [('itk_bold_to_t1', 'transforms')]),
+        ])
 
     workflow.connect([
-        (inputnode, merge_xforms, [('itk_bold_to_t1', 'in1')]),
-        (merge_xforms, bold_to_t1w_transform, [('out', 'transforms')]),
-        (inputnode, merge, [('name_source', 'header_source')]),
-        (inputnode, bold_to_t1w_transform, [('bold_split', 'input_image')]),
         (gen_ref, bold_to_t1w_transform, [('out_file', 'reference_image')]),
+        # merge 3D volumes into 4D time series
+        (inputnode, merge, [('name_source', 'header_source')]),
```

Comment:
Sorry, could you move this back above, just to reduce the diff to actually changed behavior?

Reply:
It is changed behavior here, I think -- now it's called regardless of whether MEEPI data is provided, whereas before it was only called for non-MEEPI data

Reply:
Sorry, I just meant "above  the `(gen_ref, bold_to_t1w_transform, ` line". It's identical with the removed line 322.

Reply:
ok, I can get rid of the identical comment ! sorry, I didn't realize I had duplicated that. or will removing that cause more headache for the diff ?

Reply:
Sorry, I don't understand the question. The latest update looks good, though.